src/inspector_agent.cc

```diff
@@ -554,20 +564,25 @@ void Agent::Stop() {
   if (io_ != nullptr) {
     io_->Stop();
     io_.reset();
-    enabled_ = false;
   }
 }
 
-void Agent::Connect(InspectorSessionDelegate* delegate) {
-  enabled_ = true;
-  client_->connectFrontend(delegate);
+std::unique_ptr<InspectorSession> Agent::Connect(
+    std::unique_ptr<InspectorSessionDelegate> delegate) {
+  int session_id = client_->connectFrontend(std::move(delegate));
+  return std::make_unique<InspectorSession>(session_id, client_.get());
```

Comment:
Maybe turn `client_` into a `std::shared_ptr`?

Reply:
Did that.